alphagov/pay-cardid
# PP-12222 card numbers in cardid

## WHAT YOU DID

I've added examples of the bin range data to `src/main/resources`, defaulted the configuration to look at those files, allowed the configuration to point to files within the classpath, provided worldpay's example cards (from their public documentation) into the format we expect to receive Worldpay bin ranges in.  This should allow us to overwrite the files on their way to deployment and get rid of the git submodule approach.

In theory this should not break any existing behaviour/deployments as each environment previously had to provide the location of these files and if they continue to do so the app *should* run in the same way as before.  What it allows is for us to remove that configuration and rely on the files built into the jar which will be replaced with real data on concourse.

## How to test

- Remove any configuration you have for `worldpayDataLocation`, `discoverDataLocation` and `testCardDataLocation`, it should pick up the files I've provided
- Add `worldpayDataLocation`, `discoverDataLocation` and `testCardDataLocation` back in and it should pick up the files you specify

## Code review checklist

### Logging

- [ ] Should we log the file location at start?
- [ ] Should we log when a file can't be found or is the Exception enough?

### Documentation

- [ ] When we start using this functionality (closer to the end of the epic) we'll need to update the documentation around how bin ranges are collected, processed and put live.

* Introduced default values for `worldpayDataLocation`, `discoverDataLocation` and `testCardDataLocation` which point to newly added files (based on files which already existed in this repo and information from public Worldpay sites)
